File: backend/app/schemas/file.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.file import FileIndexStatus, FileType


class IndexedFileRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    folder_id: int
    absolute_path: str
    filename: str
    extension: str
    file_type: FileType
    size_bytes: int
    checksum: str
    status: FileIndexStatus
    error_message: str | None = None
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/file.ts
export type FileType = "pdf" | "docx" | "txt" | "markdown" | "image" | "excel" | "unknown";

export type FileIndexStatus = "pending" | "extracted" | "embedded" | "failed";

export interface IndexedFile {
  id: number;
  folder_id: number;
  absolute_path: string;
  filename: string;
  extension: string;
  file_type: FileType;
  size_bytes: number;
  checksum: string;
  status: FileIndexStatus;
  error_message: string | null;
  created_at: string;
  updated_at: string;
}
